docs/design/design.md

```diff
@@ -0,0 +1,131 @@
+# Dragonfly Design Doc
+
+## Contents
+
+- [API Design](#api-design)
+  - [Reference Docs](#reference-docs)
+- [Architecture](#architecture)
+- [Workflows](#workflows)
+  - [Dfget and Supernode's communication workflows](#dfget-and-supernode's-communication-workflows)
+  - [Supernode internal component workflows](#supernode-internal-component-workflows)
+
+## Nouns/Phase Explanation
+
+- **SuperNode**: SuperNode is a long-time process with two primary responsibilities:
+  - It's the tracker and scheduler in the P2P network that choose appropriate downloading net-path for each peer.
+  - It's also a CDN server that caches downloaded data from source to avoid downloading same files repeatedly.
+- **Dfget**: Dfget is the client of Dragonfly used for downloading files. It's similar to using wget. At the same time, it also plays the role of peer, which can transfer data between each other in p2p network.
+- **Dfdaemon**: Dfdaemon establishes a proxy between container engine and registry, as a local long-running process. It filters out layer fetching requests from all requests send by container engine when pulling images, then it uses Dfget to downloading these layers.
```

Comment:
I have some comments here: Today dfdaemon is a proxy, but IMHO, it can do more, as mentioned above, dfget also plays the role of peer which needs to be running for some time(this is not a nature of cli), we probably can consider moving the peer role into dfdaemon. But honestly, what I've said may be rubbish, too. :)